timelock-server/src/integ2Test/java/com/palantir/atlasdb/timelock/AbstractMultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -264,29 +265,37 @@ public void canCreateNewClientsDynamically() {
 
     @Test
     public void lockRequestCanBlockForTheFullTimeout() {
-        LockToken token =
+        LockToken token1 =
                 client.lock(LockRequest.of(LOCKS, DEFAULT_LOCK_TIMEOUT_MS)).getToken();
-
-        Assertions.assertThatThrownBy(() -> {
-                    client.lock(LockRequest.of(LOCKS, LONGER_THAN_READ_TIMEOUT_LOCK_TIMEOUT_MS));
-                })
-                .hasCause(READ_TIMED_OUT_EXCEPTION);
-
-        client.unlock(token);
+        AtomicReference<LockToken> token2 = new AtomicReference<>(null);
+        try {
+            Assertions.assertThatThrownBy(() -> {
+                        token2.set(client.lock(LockRequest.of(LOCKS, LONGER_THAN_READ_TIMEOUT_LOCK_TIMEOUT_MS))
+                                .getToken());
+                    })
+                    .hasCause(READ_TIMED_OUT_EXCEPTION);
+        } finally {
+            client.unlock(token1);
+            if (token2.get() != null) {
+                client.unlock(token2.get());
+            }
+        }
     }
```

Comment:
Since we are setting token2 inside anon lambda, I used AtomicReference.

Reply:
I could also use optional instead of null but the latter seems more clear. I think it should be okay since this is a test code.